Infrastructure/Services/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Application.Interfaces.IRepositories;
+using Domain.Models;
+using Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Services.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : class, IEntity
+{
+    protected DataContext _context;
+    private DbSet<T?> dbSet;
+    private readonly ILogger _logger;
+    
+    public GenericRepository(DataContext context, ILogger logger)
+    {
+        _context = context;
+        this.dbSet = context.Set<T>();
+        _logger = logger;
+    }
+    
+    public virtual async Task<IEnumerable<T?>> GetAll()
+    {
+        return await dbSet.ToListAsync();
+    }
+
+    public virtual async Task<T?> GetById(int id)
+    {
+        try
+        {
+            return await dbSet.FirstOrDefaultAsync(e => e.Id == id);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error in GetById method with id: {id}", id);
+            return null;
```

Comment:
Here I see not very good practice, hiding an exception. In my opinion, if you decided to catch an exception and write in a log, let it go and raise it again: `throw;` instead of `return null;`

Reply:
fixed